### PR TITLE
fire: init at 1.0.0.3

### PR DESCRIPTION
**Pull request by OPNA2608**
###### Description of changes

Packages [Fire](https://github.com/jerryuhoo/Fire), a multi-band VST3 & AU distortion plugin.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/fire/default.nix [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  src = fetchFromGitHub {
+    owner = "jerryuhoo";
+    repo = "Fire";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-HOXkoX5CjWFdWGP8ngzPgAJ3KKSZ10aKF3CIh4rGRpw=";
+  };
+
+  patches = [
+    ./0001-Remove-FetchContent-usage.patch
```
Comment (OPNA2608) [inline pkgs/applications/audio/fire/default.nix:43]: Upstream is currently mixing Git submodules & CMake FetchContent, I've opened an issue asking them to decide on one (preferably submodules so we can just include them with `fetchSubmodules`): https://github.com/jerryuhoo/Fire/issues/46

File: pkgs/applications/audio/fire/default.nix [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  doCheck = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
+
+  # JUCE handles installation during buildPhase
+  dontInstall = true;
+
+  meta = with lib; {
+    description = "Multi-band distortion plugin by Wings";
+    homepage = "https://github.com/jerryuhoo/Fire";
+    license = licenses.agpl3Only; # Not clarified if Only or Plus
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/fire/default.nix:102]:
```suggestion
    platforms = platforms.unix;
```

File: pkgs/applications/audio/fire/default.nix [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  ];
+
+  NIX_CFLAGS_COMPILE = lib.optionalString stdenv.hostPlatform.isDarwin (toString [
+    # Fails to find fp.h on its own
+    "-isystem ${CoreServices}/Library/Frameworks/CoreServices.framework/Versions/Current/Frameworks/CarbonCore.framework/Versions/Current/Headers/"
+  ]);
+
+  doCheck = stdenv.buildPlatform.canExecute stdenv.hostPlatform;
+
+  # JUCE handles installation during buildPhase
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/fire/default.nix:95]: Could we split that or is that none trivial to do?

File: pkgs/applications/audio/fire/default.nix [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    WebKit
+    CoreServices
+    DiscRecording
+    CoreAudioKit
+  ];
+
+  NIX_CFLAGS_COMPILE = lib.optionalString stdenv.hostPlatform.isDarwin (toString [
+    # Fails to find fp.h on its own
+    "-isystem ${CoreServices}/Library/Frameworks/CoreServices.framework/Versions/Current/Frameworks/CarbonCore.framework/Versions/Current/Headers/"
+  ]);
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/fire/default.nix:91]:
```suggestion
  # Fails to find fp.h on its own
  NIX_CFLAGS_COMPILE = lib.optionalString stdenv.hostPlatform.isDarwin "-isystem ${CoreServices}/Library/Frameworks/CoreServices.framework/Versions/Current/Frameworks/CarbonCore.framework/Versions/Current/Headers/";
```

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise LGTM

File: pkgs/applications/audio/fire/default.nix (hunk 2 above)
Comment (OPNA2608) [reply pkgs/applications/audio/fire/default.nix:102]: `all` isn't accurate cuz of all the embedded targets in there which have no use for VST3s, but neither is just `unix` if we're honest… meh.

File: pkgs/applications/audio/fire/default.nix (hunk 3 above)
Comment (OPNA2608) [reply pkgs/applications/audio/fire/default.nix]: It's abit messier IMO, but done.

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -27532,6 +27532,10 @@ with pkgs;
 
   filezilla = callPackage ../applications/networking/ftp/filezilla { };
 
+  fire = callPackage ../applications/audio/fire {
+    inherit (darwin.apple_sdk.frameworks) Cocoa WebKit CoreServices DiscRecording CoreAudioKit;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:27536]:
```suggestion
  fire = darwin.apple_sdk_11_0.callPackage ../applications/audio/fire {
    inherit (darwin.apple_sdk_11_0.frameworks) Cocoa WebKit CoreServices DiscRecording CoreAudioKit;
```
maybe that fixes the darwin build failure already

Comment (OPNA2608) [reply pkgs/top-level/all-packages.nix]: Also needed some additional dependencies, fixed now.

**Comment by gador** [comment]
Result of `nixpkgs-review pr 188515` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fire</li>
  </ul>
</details>

**Pull request merged**